pkgs/os-specific/linux/firmware/fwupd/default.nix

```diff
@@ -203,6 +209,11 @@ stdenv.mkDerivation rec {
     "--localstatedir=/var"
     "--sysconfdir=/etc"
     "-Dsysconfdir_install=${placeholder "out"}/etc"
+
+    # We do not want to place the daemon into lib
+    "--libexecdir=${placeholder "out"}/libexec"
+    # Our builder only adds $lib/lib into rpath
+    "-Dc_link_args=-Wl,-rpath,${placeholder "out"}/lib"
```

Comment:
Everything else I think I understand here, but would you mind explaining this for me?

Reply:
Since Nix does not have `/usr/lib`, every library a program depends on must have path for finding it listed in `DT_RUNPATH` entry in the ELF file. Our generic builder populates the entry for us but apparently only with `lib` outputs. Because the `libfwupdplugin` is in `out` its location is not included so we need to add it manually.

I believe the builder adds every `/lib` directory to entry of every<sup>[citation needed]</sup> ELF file but superfluous paths (those not containing libraries listed in `DT_NEEDED` entries) are stripped by<sup>[citation needed]</sup> `strip` setup-hook so there is no cycle.

Reply:
I improved the comments slightly.

Reply:
Thanks a lot :+1:  All clear here.